### PR TITLE
Update gossipsub

### PR DESCRIPTION
**Pull request by wemeetagain**
- Update gossipsub to v0.7.0
- Update msg id to be 1.0 compatible (and also compatible with newest gossipsub)
- Update gossip mesh params to be 1.0 compatible
- Minor tweaks to lodestar gossipsub (cache msg id, override different methods, etc)

### CODE REVIEW

**Review by 3xtr4t3rr3str14l — APPROVED** [review]
LGTM

File: packages/lodestar/src/network/gossip/gossipsub.ts [hunk 1]
```diff
@@ -141,24 +144,46 @@ export class LodestarGossipsub extends Gossipsub {
...
+    if (encoding === GossipEncoding.SSZ_SNAPPY) {
+      try {
+        const uncompressed = uncompress(data);
+        return hash(Buffer.concat([MESSAGE_DOMAIN_VALID_SNAPPY, uncompressed])).slice(0, 20);
+      } catch (e) {
+        return hash(Buffer.concat([MESSAGE_DOMAIN_INVALID_SNAPPY, data])).slice(0, 20);
+      }
     }
-    return super._publish(message);
+    return hash(data).slice(0, 20);
```
Comment (dapplion) [inline packages/lodestar/src/network/gossip/gossipsub.ts:186]: make this `20` a variable such as `GOSSIP_SUB_MSG_ID_LENGTH`

File: packages/lodestar/src/network/gossip/gossipsub.ts [hunk 2]
```diff
@@ -194,13 +219,14 @@ export class LodestarGossipsub extends Gossipsub {
     return result as GossipMessageValidatorFn;
   }
 
-  private deserializeGossipMessage(topic: string, message: InMessage): {object: GossipObject; subnet?: number} {
+  private deserializeGossipMessage(topic: string, msg: InMessage): {object: GossipObject; subnet?: number} {
+    let data = msg.data!;
     if (getTopicEncoding(topic) === GossipEncoding.SSZ_SNAPPY) {
-      message.data = uncompress(message.data!);
+      data = uncompress(data);
```
Comment (dapplion) [inline packages/lodestar/src/network/gossip/gossipsub.ts:225]: Mutation can be prevented with
```ts
const data = getTopicEncoding(topic) === GossipEncoding.SSZ_SNAPPY
  ? uncompress(data)
  : data
```

File: packages/lodestar/src/network/gossip/gossipsub.ts (hunk 1 above)
Comment (dapplion) [reply packages/lodestar/src/network/gossip/gossipsub.ts:186]: Would make sense to add it to `packages/lodestar/src/network/gossip/constants.ts` ?

**Comment by dapplion** [comment]
Applied review comments in new PR https://github.com/ChainSafe/lodestar/pull/1781

**Pull request merged**